cmd/ipfs/daemon.go

```diff
@@ -461,3 +482,80 @@ func merge(cs ...<-chan error) <-chan error {
 	}()
 	return out
 }
+
+func runGC(req cmds.Request, repo repo.Repo, node *core.IpfsNode) error {
```

Comment:
@rht  after parsing options and figuring out inputs, is there anything preventing this from moving into the `repo` package?

the majority of this routine belongs inside of the `repo` package. what we can do is make a function called:

```
func PerioficGC(repo Repo) error { ... }
```

the reasons for want it in repo and not in daemon are:
- it may be called by nodes who embed IPFS as a library (instead of running their own daemon)
- it belongs there as it is repo-specific and daemon logic does not need to complicate itself with this

the repo should already have the cfg, so has access to `StorageMax`, and `StrageGCWatermark`. 
and, it looks like this same routine should work for all repos, which support GC. (S3 Datastore/Repo)

cc @whyrusleeping